src/engine/strat_engine/serde_structs.rs

```diff
@@ -71,6 +73,36 @@ pub trait Recordable<T: Serialize> {
     fn record(&self) -> T;
 }
 
+/// List of optional features for pools.
+#[derive(Debug, Deserialize, Eq, PartialEq, Serialize, Hash, Copy, Clone)]
+pub enum PoolFeatures {
+    Raid,
+    Integrity,
+    Encryption,
+}
+
+/// Features enabled on the given pool to be stored in the metadata.
+#[derive(Debug, Deserialize, Eq, PartialEq, Serialize)]
```

Comment:
This is the first use of a ```HashSet``` for our Serde structs. With thin pool features args, which are also really conceptually a set, we went with a ```Vec<String>``` (reason unknown). In our Record implementation for ```ThinPoolDevSave``` we alphabetically sort the feature args, so that a textual diff won't make them appear to have changed when they haven't. It would be helpful to make sure that the PoolFeatures are also sorted in a canonical way, probably alphabetically, when they are written.  For this reason, I think it would be reasonable to use a ```Vec``` instead, since the features end up in arbitrary order in the HashSet and then the way to write them ordered is probably some ```serialize_with``` annotation and that seems like a lot of bits and pieces for limited gain.

Reply:
Resolved.